### PR TITLE
Document.onload

### PR DESCRIPTION
**Pull request by bensonnn**

### CODE REVIEW

File: src/study.js [hunk 1]
```diff
@@ -89,6 +90,13 @@ class Study {
    */
   assign(testName, bucketName) {
     const oldAssignments = this.assignments();
+    const applyClasses = () => {
+      Object.keys(this.classList).forEach((name) => {
+        if (this.classList[name]) document.body.classList.add(`${name}--${this.classList[name]}`);
```
Comment (jakiestfu) [inline src/study.js]: Should we do a `hasOwnProperty` check here?

File: src/study.js [hunk 2]
```diff
@@ -44,6 +44,7 @@ class Study {
     this.userBuckets = {};
     this.userAssignments = {};
     this.providedTests = [];
+    this.classList = {};
```
Comment (yowainwright) [inline src/study.js]: is classList an `array`?

Comment (yowainwright) [inline src/study.js]: Nevermind, I see what you did (line 93). 🙌

File: src/study.js [hunk 3]
```diff
@@ -89,6 +90,13 @@ class Study {
    */
   assign(testName, bucketName) {
     const oldAssignments = this.assignments();
+    const applyClasses = () => {
+      Object.keys(this.classList).forEach((name) => {
+        if (this.classList[name]) document.body.classList.add(`${name}--${this.classList[name]}`);
+        else document.body.classList.remove(`${name}--${oldAssignments[name]}`);
+      });
+      document.removeEventListener('DOMContentLoaded', applyClasses);
```
Comment (yowainwright) [inline src/study.js]: I don't know if `'DOMContentLoaded'` is reliable ...? What about an observer?

File: src/study.js [hunk 4]
```diff
@@ -125,15 +133,18 @@ class Study {
...
       } else {
         // Add to our assignments
-        if (!isServer) { document.body.classList.add(`${test.name}--${bucket}`); }
+        this.classList[test.name] = bucket;
         this.userAssignments[test.name] = bucket;
       }
     });
 
+    if (!isServer && document.body) applyClasses();
+    else document.addEventListener('DOMContentLoaded', applyClasses);
```
Comment (jakiestfu) [inline src/study.js]: What @yowainwright said. Is this fully supported across the browsers we support?

File: src/study.js (hunk 3 above)
Comment (yowainwright) [inline src/study.js]: Thanks for explaining. I think the observer is a non-issue.

**Pull request merged**